R/findNLDI.R

```diff
@@ -0,0 +1,113 @@
+#' @title Find Linked Hydro Data via the NLDI
+#'
+#' @description This function provides a formal query to the Network Linked Data Index (NLDI).
+#' There are three tasks that can be accomplished (1) feature discovery (2) NHD Network Tracing (3) Network entity discovery.
+#' One parameter ('comid', 'nwis', 'np_des', or 'wqp') must be supplied to indicate where to start on NHD
+#' The 'nav' parameter describes what direction the network should be traced from the starting point (UT = Upper Tributary, UM = Upper Mainsteam, DD = Down Stream Diversion, DM = Down Stream Mainstem)
+#' Finally, the 'find' parameter, describes what features should be retunred. The 'km' parameter can limit the distance the network is traced.
+#'
+#' @param comid an NHD COMID
+#' @param nwis  a USGS NWIS station ID
+#' @param huc12 a HUC12 ID
+#' @param npdes_rad Facilities that Discharge to Water NHDPlus Indexed Dataset (ID)
+#' @param wqp a water quality portal location ID
+#' @param nav where to navigate from the starting point
+#' @param find what resources to find along the navigation path ("comid", "huc12pp", "npdes_rad", "nwis", "wqp", "basin")
+#' @param distance_km how far to look along the navigation path (in kilometers)
+#' @return a list of sf features
+#' @importFrom httr RETRY
+#' @importFrom sf read_sf
+#' @export
+#' @examples
+#' \donttest{
+#' # Starting at NWIS 05428500, navigate along the upper mainsteam, and upper tributary
+#' #to find the basin and NWIS sites:
+#'
+#'  nldi = findNLDI(nwis = "05428500", nav = c("UT", "UM", "DM"), find = c("nwis", "basin"))
+#'
+#'  plot(nldi$basin$geometry, lwd = 2, col = NA, border = 'black')
+#'  plot(nldi$UT$geometry, col = 'blue', add = T)
+#'  plot(nldi$DM$geometry, col = 'purple', add = T)
+#'  plot(nldi$UM$geometry, col = 'red', lwd = 3, add = T)
+#'  plot(nldi$nwissite$geometry, pch = 16, col = 'darkgreen', add = T)
+#' }
+
+findNLDI = function(comid = NULL, nwis = NULL, huc12 = NULL, npdes_rad = NULL, wqp = NULL,
+                    nav = NULL, find = NULL, distance_km = NULL) {
+
+  get_nldi = function(url){
```

Comment:
Should use query nldi. https://github.com/USGS-R/nhdplusTools/blob/master/R/get_nldi.R#L170